app/src/main/java/org/thoughtcrime/securesms/contacts/ContactAccessor.java

```diff
@@ -0,0 +1,340 @@
+    };
+
+    public final String number;
+    public final String type;
+
+    public NumberData(String type, String number) {
+      this.type = type;
+      this.number = number;
+    }
+
+    public NumberData(Parcel in) {
+      number = in.readString();
+      type   = in.readString();
+    }
+
+    public int describeContents() {
+      return 0;
+    }
+
+    public void writeToParcel(Parcel dest, int flags) {
+      dest.writeString(number);
+      dest.writeString(type);
+    }
+  }
+
+  public static class ContactData implements Parcelable {
+
+    public static final Parcelable.Creator<ContactData> CREATOR = new Parcelable.Creator<ContactData>() {
+      public ContactData createFromParcel(Parcel in) {
+        return new ContactData(in);
+      }
+
+      public ContactData[] newArray(int size) {
+        return new ContactData[size];
+      }
+    };
+
+    public final long id;
+    public final String name;
+    public final List<NumberData> numbers;
+
+    public ContactData(long id, String name) {
+      this.id      = id;
+      this.name    = name;
+      this.numbers = new LinkedList<NumberData>();
+    }
+
+    public ContactData(Parcel in) {
+      id      = in.readLong();
+      name    = in.readString();
+      numbers = new LinkedList<NumberData>();
+      in.readTypedList(numbers, NumberData.CREATOR);
+    }
+
+    public int describeContents() {
+      return 0;
+    }
+
+    public void writeToParcel(Parcel dest, int flags) {
+      dest.writeLong(id);
+      dest.writeString(name);
+      dest.writeTypedList(numbers);
+    }
+  }
+
+  /***
+   * If the code below looks shitty to you, that's because it was taken
+   * directly from the Android source, where shitty code is all you get.
+   */
+
+  public Cursor getCursorForRecipientFilter(CharSequence constraint,
+      ContentResolver mContentResolver)
+  {
+    final String SORT_ORDER = Contacts.TIMES_CONTACTED + " DESC," +
+                              Contacts.DISPLAY_NAME + "," +
+                              Contacts.Data.IS_SUPER_PRIMARY + " DESC," +
+                              Phone.TYPE;
+
+    final String[] PROJECTION_PHONE = {
+        Phone._ID,                  // 0
+        Phone.CONTACT_ID,           // 1
+        Phone.TYPE,                 // 2
+        Phone.NUMBER,               // 3
+        Phone.LABEL,                // 4
+        Phone.DISPLAY_NAME,         // 5
+    };
+
+    String phone = "";
+    String cons  = null;
+
+    if (constraint != null) {
+      cons = constraint.toString();
+
+      if (RecipientsAdapter.usefulAsDigits(cons)) {
+        phone = PhoneNumberUtils.convertKeypadLettersToDigits(cons);
+        if (phone.equals(cons) && !PhoneNumberUtils.isWellFormedSmsAddress(phone)) {
+          phone = "";
+        } else {
+          phone = phone.trim();
+        }
+      }
+    }
+    Uri uri = Uri.withAppendedPath(Phone.CONTENT_FILTER_URI, Uri.encode(cons));
+    String selection = String.format("%s=%s OR %s=%s OR %s=%s",
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format int: Phone.TYPE_WORK_MOBILE, Phone.TYPE_MMS, Phone.TYPE_MOBILE. This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 300. Consider formatting this data with &#39;%d&#39; instead.